src/renderer/gdi/paint.cpp

```diff
@@ -445,10 +456,42 @@ using namespace Microsoft::Console::Render;
         for (size_t i = 0; i != _cPolyText; ++i)
         {
             const auto& t = _pPolyText[i];
-            if (!ExtTextOutW(_hdcMemoryContext, t.x, t.y, t.uiFlags, &t.rcl, t.lpstr, t.n, t.pdx))
+
+            // The following if/else replicates the essentials of how ExtTextOutW() without ETO_IGNORELANGUAGE works.
+            // See InternalTextOut().
+            //
+            // Unlike the original, we don't check for `GetTextCharacterExtra(hdc) != 0`,
+            // because we don't ever call SetTextCharacterExtra() anyways.
+            //
+            // GH#12294:
+            // Additionally we set ss.fOverrideDirection to TRUE, because we need to present RTL
+            // text in logical order in order to be compatible with applications like `vim -H`.
+            if (_fontHasWesternScript && ScriptIsComplex(t.lpstr, t.n, SIC_COMPLEX) == S_FALSE)
             {
-                hr = E_FAIL;
-                break;
+                if (!ExtTextOutW(_hdcMemoryContext, t.x, t.y, t.uiFlags | ETO_IGNORELANGUAGE, &t.rcl, t.lpstr, t.n, t.pdx))
+                {
+                    hr = E_FAIL;
+                    break;
+                }
+            }
+            else
+            {
+                SCRIPT_STATE ss{};
+                ss.fOverrideDirection = TRUE;
+
+                SCRIPT_STRING_ANALYSIS ssa;
+                hr = ScriptStringAnalyse(_hdcMemoryContext, t.lpstr, t.n, 0, -1, SSA_GLYPHS | SSA_FALLBACK, 0, nullptr, &ss, t.pdx, nullptr, nullptr, &ssa);
+                if (FAILED(hr))
+                {
+                    break;
+                }
+
+                hr = ScriptStringOut(ssa, t.x, t.y, t.uiFlags, &t.rcl, 0, 0, FALSE);
+                std::ignore = ScriptStringFree(&ssa);

```

Comment:
clever! is std::ignore supported in the version of C++ that Windows builds with?

Reply:
keep in mind thsi may be backported to earlier versions of Windows, stuck on c++17

Reply:
Yeah `std::ignore` is C++11.